src/vaults.lua

```diff
@@ -74,10 +86,28 @@ function vaults.vaultedTransfer(from, recipient, qty, lockLengthMs, currentTimes
 		balance = qty,
 		startTimestamp = currentTimestamp,
 		endTimestamp = currentTimestamp + lockLengthMs,
+		controller = revokable and from or nil,
 	})
 	return newVault
 end
 
+--- Revokes a vaulted transfer
+---@param from WalletAddress
+---@param recipient WalletAddress
+---@param vaultId VaultId
+---@param currentTimestamp Timestamp
+---@return Vault
+function vaults.revokeVaultedTransfer(from, recipient, vaultId, currentTimestamp)
+	local vault = vaults.getVault(recipient, vaultId)
+	assert(vault, "Vault not found.")
+	assert(vault.controller == from, "Only the controller can revoke the vault.")
+	assert(currentTimestamp <= vault.endTimestamp, "Vault has ended.")
+
+	balances.increaseBalance(recipient, vault.balance)
```

Comment:
The balance is supposed to go to the `from`! This was my concern about the ambiguity in the argument names.

Reply:
yikes, great catch

fixed unit coverage here, unluckily I used an address that already has 100 balance in the test env: [1835ad6](https://github.com/ar-io/ar-io-network-process/pull/334/commits/1835ad63b2aa4dead92ec60fc1850a6fcc55b4f9)

actual fix here for red/green in CI: [c8ed870](https://github.com/ar-io/ar-io-network-process/pull/334/commits/c8ed870db47188043ce463ff144ff6b97a18401a)

![image](https://github.com/user-attachments/assets/518f1ca1-ec6b-4858-998e-5834d979eecc)